File: src/types/tag.ts
export interface Tag {
  id: number;
  name: string;
  description: string | null;
  ignore_auto_tag: boolean;
  favorite: boolean;
  created_at: string;
  updated_at: string;
}

export interface TagCreate {
  name: string;
  description?: string;
  ignore_auto_tag?: boolean;
  favorite?: boolean;
}

File: src-tauri/src/models/tag.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Tag {
    pub id: i64,
    pub name: String,
    pub description: Option<String>,
    pub ignore_auto_tag: bool,
    pub favorite: bool,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TagCreate {
    pub name: String,
    pub description: Option<String>,
    pub ignore_auto_tag: Option<bool>,
    pub favorite: Option<bool>,
}
